chmlib fixes, adds and misc

glachancecmaisonneuve opened the pull request:
Fixed: debug and release versions in their proper folder (was reversed)
Added: Tools included in that library
Misc: portfile cleanup, chm.vcxproj optimization switches

glachancecmaisonneuve committed:
Fixed: debug and release versions in their proper folder (was reversed)
Added: Tools included in that library
Misc: portfile cleanup, chm.vcxproj optimization switches

msftclas commented:
@glachancecmaisonneuve,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by Microsoft. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

alexkaratarakis commented on ports/chmlib/portfile.cmake:
```diff
@@ -1,26 +1,5 @@
-
 if(${VCPKG_LIBRARY_LINKAGE} STREQUAL "dynamic")
-    message(FATAL_ERROR "chmlib supports static linking only.")
+    message(FATAL_ERROR "CHMLIB SUPPORTS STATIC LINKING ONLY.")
```
Instead of failing the build you could warn that dynamic is not supported and override the setting to build as static. Many portfiles do this.

```
if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
    message(STATUS "Warning: Dynamic building not supported yet. Building static.")
    set(VCPKG_LIBRARY_LINKAGE static)
endif()
```
(Not sure if the "yet" is appropriate or should be removed)

This way you can get chmlib even in dynamic triplets, instead of getting an error.

glachancecmaisonneuve replied:
good idea

alexkaratarakis commented:
Looks good!
Please also bump the version in the `CONTROL` file : 0.40 -> 0.40-1.
This is to signal users of the library that there is an update.

glachancecmaisonneuve committed:
Dynamic build -> Static without failing the build

glachancecmaisonneuve committed:
bump subversion

glachancecmaisonneuve commented:
That last push should do it.  
Summary of changes
- Fixed: debug and release versions in their proper folder (was reversed)
- Added: Tools included in that library
- Misc: portfile cleanup, chm.vcxproj optimization switches
- CONTROL file : 0.40 -> 0.40-1.
- DLL build is STATIC

alexkaratarakis committed:
[chmblib] Also modify VCPKG_LIBRARY_LINKAGE

alexkaratarakis commented:
I also added `set(VCPKG_LIBRARY_LINKAGE static)` so that the post-build checker performs the right checks.
Thanks for the PR!
